### PR TITLE
Languages.hcl terraform semantic highlights

### PR DESCRIPTION
**Pull request by lkishalmi**
This change removes the mostly redundant TerraformLexer grammar, which added a few keywords of Terraform dialect of HCL.

The coloring is done by the real Terraform specific structure analyzer. There are room for improvements, especially when I'd finish the expression part of the AST, but let's do baby steps.

### CODE REVIEW

File: ide/languages.hcl/src/org/netbeans/modules/languages/hcl/terraform/TerraformParserResult.java [hunk 1]
```diff
@@ -116,8 +122,11 @@ protected void processDocument(HCLDocument doc, SourceRef references) {
             } else {
                 references.getOffsetRange(type).ifPresent((range) -> addError(Bundle.UNKNOWN_BLOCK(type.id()), range));
             }
+            checkDuplicateAttribute(doc, references);
```
Comment (matthiasblaesing) [inline ide/languages.hcl/src/org/netbeans/modules/languages/hcl/terraform/TerraformParserResult.java:125]: This leads to errors reported multiple times. This either needs to be moved out of the loop (started in new line 101) or doc must become `block`:

```suggestion
            checkDuplicateAttribute(block, references);
```

File: ide/languages.hcl/src/org/netbeans/modules/languages/hcl/FontAndColors.xml [hunk 2]
```diff
@@ -36,4 +36,8 @@
     <fontcolor name="separator" default="separator"/>
     <fontcolor name="string" default="string"/>
     <fontcolor name="whitespace" default="whitespace"/>
+
+    <fontcolor name="mod-method" default="keyword">
+        <font style="bold"/>
+    </fontcolor>
```
Comment (matthiasblaesing) [inline ide/languages.hcl/src/org/netbeans/modules/languages/hcl/FontAndColors.xml:42]: I notice this:

```
SEVERE [org.netbeans.modules.csl.editor.semantic.ColoringManager]: no colors for: mod-class with mimetext/x-terraform+x-hcl
```

`mod-class` also needs to be added to the colors definition:

```suggestion
    <fontcolor name="mod-method" default="keyword">
        <font style="bold"/>
    </fontcolor>
    
    <fontcolor name="mod-class" default="keyword">
        <font style="bold"/>
    </fontcolor>
```

File: ide/languages.hcl/src/org/netbeans/modules/languages/hcl/terraform/TerraformParserResult.java (hunk 1 above)
Comment (lkishalmi) [reply ide/languages.hcl/src/org/netbeans/modules/languages/hcl/terraform/TerraformParserResult.java]: Valid.

File: ide/languages.hcl/src/org/netbeans/modules/languages/hcl/FontAndColors.xml (hunk 2 above)
Comment (lkishalmi) [reply ide/languages.hcl/src/org/netbeans/modules/languages/hcl/FontAndColors.xml]: Played around wit these modifiers, checked in a wrong version. Fixed now.

**Comment by lkishalmi** [comment]
So added a few features in the last commit:
* Added a group separator coloring option for `{}` and `[]`
* Terraform Declaration blocks (I could not find the correct term for root level blocks) got recognized, their ID is bold now
* Terraform attribute names highlighted as fields
* Added Tabs and Insert formatting to the Options dialog, unfortunately does not really work without a formatter.
* Added code samples to the Options dialog

**Comment by lkishalmi** [comment]
image

**Review by matthiasblaesing — APPROVED** [review]
Looks sane to me.

**Pull request merged**